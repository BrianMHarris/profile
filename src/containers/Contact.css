.contact-content {
  display: flex;
  flex-direction: column;
  align-items: center;
  box-sizing: border-box;
  padding: 100px 10% 100px 10%;
  font-family: 'Poppins', sans-serif;
  color: var(--clr-txt-main);
}

.contact-flex {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  height: 100%;
}

.contact-body p,
.contact-form p {
  font-size: 1.1em;
  text-align: center;
  max-width: 450px;
}

.contact-title {
  font-size: 1.75em;
  text-transform: uppercase;
  text-align: center;
  width: 100%;
  color: var(--clr-txt-standout);
  font-family: 'Nunito', sans-serif;
}

.contact-body {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.contact-email {
  display: flex;
  justify-content: center;
  margin: 20px 0 24px 0;
}

.contact-email a:link {
  text-decoration: none;
}

@media screen and (min-width: 361px) and (max-width: 767px) {
  .contact-title {
    font-size: 2.25em;
  }

  .contact-body p,
  .contact-form p {
    font-size: 1.25em;
  }
}

@media screen and (min-width: 768px) {
  .contact-title {
    font-size: 2.5em;
  }

  .contact-body p,
  .contact-form p {
    font-size: 1.5em;
  }
}

@media screen and (min-width: 1100px) {
  .contact-flex {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-start;
  }

  .contact-body {
    width: 46%;
    height: 100%;
    margin: 0;
  }

  .contact-form {
    width: 46%;
    height: 100%;
  }
}
