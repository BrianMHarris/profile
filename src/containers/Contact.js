import React, { Component } from "react"
import "./Contact.css";
import ContactForm from "./ContactForm";
import ButtonNoScroll from '../components/ButtonNoScroll'

class Contact extends Component {
  render() {
    return (
      <div className="contact-content">
        <div id="contact"></div>
        <div className="contact-title">
          contact me
        </div>
        <div className="contact-flex">
          <div className="contact-body">
            <p>
              If you have an inquiry or would like to speak with me about a project:
            </p>
            <div className="contact-email">
              <a href="mailto:brianharris.dev@gmail.com">
                <ButtonNoScroll addClasses="btnns-hvr btnns-contact btnns-large"
                      label="Send an Email"
                />
              </a>
            </div>
          </div>
          <div className="contact-form">
            <p>
              or fill out this handy form:
            </p>
            <ContactForm />
          </div>
        </div>
      </div>

    );
  }
}

export default Contact;
