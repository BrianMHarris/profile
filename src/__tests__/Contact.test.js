import React from 'react';
import ReactDOM from 'react-dom';
import Contact from '../containers/Contact';
import renderer from "react-test-renderer";

describe('Contact', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Contact />,
      div);
  });

  it('matches the snapshot', () => {
    const component = renderer.create(
      <Contact />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
