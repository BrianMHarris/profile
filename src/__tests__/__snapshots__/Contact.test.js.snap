// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact matches the snapshot 1`] = `
<div
  className="contact-content"
>
  <div
    id="contact"
  />
  <div
    className="contact-title"
  >
    contact me
  </div>
  <div
    className="contact-flex"
  >
    <div
      className="contact-body"
    >
      <p>
        If you have an inquiry or would like to speak with me about a project:
      </p>
      <div
        className="contact-email"
      >
        <a
          href="mailto:brianharris.dev@gmail.com"
        >
          <button
            className="btnns-std btnns-hvr btnns-contact btnns-large"
            onClick={null}
            onMouseOut={undefined}
            onMouseOver={undefined}
            type={undefined}
          >
            Send an Email
          </button>
        </a>
      </div>
    </div>
    <div
      className="contact-form"
    >
      <p>
        or fill out this handy form:
      </p>
      <div
        className="cf-form"
      >
        <div
          className="cf-body"
        />
        <form
          action="#"
          onSubmit={[Function]}
        >
          <div
            className="cf-inputs-wrap"
          >
            <input
              className="cf-input-name"
              id="cf-fullName"
              name="fullName"
              onChange={[Function]}
              placeholder="name"
              required={true}
              type="text"
              value=""
            />
            <input
              className="cf-input-email"
              name="email"
              onChange={[Function]}
              placeholder="email"
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="cf-inputs-fixed"
          >
            <textarea
              className="cf-input-large cf-input-body"
              name="body"
              onChange={[Function]}
              placeholder="message"
              required={true}
              type="text"
              value=""
            />
          </div>
          <button
            className="btnns-std btnns-hvr btnns-contact btnns-nomargins"
            onClick={null}
            onMouseOut={undefined}
            onMouseOver={undefined}
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
